File: RestAPIs/Admin_Services/Models/Bill.cs
using System;
using System.Collections.Generic;

namespace Admin_Services.Models;


public partial class Bill
{
    public int BillId { get; set; }

    public int UserId { get; set; }

    public int ServiceId { get; set; }

    public decimal Amount { get; set; }

    public decimal? Tax { get; set; }

    public decimal TotalAmount { get; set; }

    public DateOnly? BillDate { get; set; }

    public string PaymentMode { get; set; } = null!;

    public virtual Service Service { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: User Services/freelancer/src/main/java/com/worksphere/freelancer/entity/Bill.java
package com.worksphere.freelancer.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "bill")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Bill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bill_id")
    private Integer billId;

    @Column(name = "user_id")
    private Integer userId;

    @Column(name = "freelancer_id")
    private Integer freelancerId;

    // service_id -> services.service_id
    @ManyToOne
    @JoinColumn(name = "service_id")
    private Services service;

    @ManyToOne
    @JoinColumn(name = "requirement_id")
    private Requirement requirement;

    @Column(name = "amount", precision = 10, scale = 2)
    private BigDecimal amount;

    @Column(name = "tax", precision = 10, scale = 2)
    private BigDecimal tax;

    @Column(name = "total_amount", precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @Column(name = "bill_date")
    private LocalDate billDate;

    @Column(name = "payment_mode")
    private String paymentMode;

    @Column(name = "payment_status")
    private String paymentStatus;

    @Column(name = "transaction_id")
    private String transactionId;

    @Column(name = "paid_date")
    private LocalDate paidDate;
}
